microprofile/WFLY-10712_microprofile_metrics.asciidoc

```diff
@@ -0,0 +1,230 @@
+# TYPE base:cpu_system_load_average gauge
+base:cpu_system_load_average 2.3134765625
+...
+# HELP vendor:foo <description>
+# TYPE vendor:foo <type>
+vendor:foo 12345.0
+----
+
+To fetch metrics in the JSON format, the `Accept` HTTP header MUST be set to `application/json`:
+
+----
+$ curl -v -H "Accept: application/json" http://127.0.0.1:9990/metrics/
+...
+< HTTP/1.1 200 OK
+< Content-Type: application/json
+...
+{"base" :
+{
+  "classloader.totalLoadedClass.count" : 10911,
+  "cpu.systemLoadAverage" : 2.1201171875,
+  ...
+}
+,"vendor" :
+{
+  ...
+  "foo": 12345.0
+}
+----
+
+==== Authentication
+
+By default, the __HTTP endpoints require authentication__. This can be alleviated by explicitly setting
+the `security-enabled` of the `/subsystem=microprofile-smallrye-metrics` to `false`.
+
+If security is enabled, the HTTP client must be authenticated (otherwise, the server will reply with a
+  `401 NOT AUTHORIZED` response):
+
+----
+$ curl -v http://127.0.0.1:9990/metrics/
+...
+< HTTP/1.1 401 Unauthorized
+----
+
+In that case, WildFly MUST have a management user and the HTTP client MUST pass its credential to the Metrics HTTP endpoint:
+
+----
+$ curl -v --digest -u admin:adminpwd http://127.0.0.1:9990/metrics
+< HTTP/1.1 200 OK
+...
+# HELP base:cpu_system_load_average Displays the system load average for the last minute. The system load average is the sum of the number of runnable entities queued to the available processors and the number of runnable entities running on the available processors averaged over a period of time. The way in which the load average is calculated is operating system specific but is typically a damped time-dependent average. If the load average is not available, a negative value is displayed. This attribute is designed to provide a hint about the system load and may be queried frequently. The load average may be unavailable on some platform where it is expensive to implement this method.
+# TYPE base:cpu_system_load_average gauge
+base:cpu_system_load_average 1.9658203125
+...
+----
+
+If security is disabled in the subsystem configuration, the HTTP client does not require authentication (and it does not require a WildFly management user either).
+
+Note that the _standalone profiles will explicitly disable authentication_ as the HTTP management interface is bound to `127.0.0.1`.
+
```

Comment:
something special for https ?

Reply:
No. 
The extension requires only the http management interface and it is not possible to bind it to the https management interface (that's the design already chosen for MicroProfile Health Check)

Reply:
No. 
The extension requires only the http management interface and it is not possible to bind it to the https management interface (that's the design already chosen for MicroProfile Health Check)

Reply:
You're right, I've removed the justification being that the management interface is bound to `127.0.0.1`.